Created person card component which is also using the Avatar component

Noorieee opened the pull request:
Advent CSS design used as a base / reference: 
![image](https://github.com/user-attachments/assets/b53af19b-c5fb-4e7d-8bd8-598c60d37afb)

My component:
![image](https://github.com/user-attachments/assets/488ff536-dd48-404a-bc92-3ad6c6f5aa7d)
![image](https://github.com/user-attachments/assets/568b0dd2-89f9-4098-b21f-7f14d4c438fe)
![image](https://github.com/user-attachments/assets/c1440e28-9403-4ec6-a388-23921094898b)
![image](https://github.com/user-attachments/assets/fb7663c9-0445-459e-8734-5e29ffe9a156)
![image](https://github.com/user-attachments/assets/ddd102f5-cc76-4bbf-9f57-ad0cef8c86cd)

Noorieee committed:
Created person card component which is also using the Avatar component

Noorieee committed:
Changes made:
Increased min width on the HiddenDetail styled component
Added missing indicator prop in the interface, type and on the AvatarContainer
Adjusted stories to hold indicator and added defaults for optional props

goralight commented on src/components/Molecules/PersonCard/index.tsx:
```diff
@@ -0,0 +1,126 @@
+type Indicator = 'online' | 'away' | 'busy'
+
+interface PersonCardProps {
+  name: string
```
if name and email is mandatory - but `isHidden` is a thing, would it not make sense to just make the email and name optional, but throw an error if we dont have at least name and email OR isHidden

Noorieee replied:
In this case name and email should be mandatory on the card as we would always want both of these pieces of information. Do you still recommend adding the error?

goralight commented on src/components/Molecules/PersonCard/index.tsx:
```diff
@@ -0,0 +1,126 @@
+  vanity?: boolean
+  indicator?: Indicator
+  isHidden?: boolean
+  initial: string
```
If we have the `name` why do we need an `initial` prop, cant we figure that out from the name?

Noorieee replied:
That is smart and makes sense! Updated my code with this now

goralight commented on src/components/Molecules/PersonCard/index.tsx:
```diff
@@ -0,0 +1,126 @@
+const PersonCard = ({ name, email, imageUrl, indicator, vanity, isHidden, initial }: PersonCardProps) => {
+  return (
+    <PersonCardContainer>
+      <AvatarContainer><Avatar initial={initial} imageUrl={imageUrl} indicator={indicator} isHidden={isHidden} vanity={vanity} /></AvatarContainer>
```
new line plz

Noorieee committed:
Changes made:
Removed the initial prop and added a variable that gets the initial from name instead
Removed initial from stories

Noorieee committed:
Added more defence to Avatar and PersonCard props
Avatar changes:
Updated some comments to help explain bits of the code
Made intial prop optional and gave it a default value of an empty string
Added a check that throws an error if initial is empty, isHidden is false and imageUrl is empty
Updated if statement that throws an error if a number is added as an initial by also checking if initial exists also
Updated stories and added the default value for intial as empty string
PersonCard changes:
Made name and email optional
Added a check that throws an error if name and email are empty and isHidden is false
Updated some comments to help explain bits of the code
Added optional chaining to check if name exists then get the first character from he name and stores that into initial / sets as initial
Updated stories and removed required from name and email

Noorieee committed:
Merge branch 'main' into person-card-comp

goralight commented:
the size is different for isHidden view and !isHidden view

goralight commented on src/components/Molecules/PersonCard/index.tsx:
```diff
@@ -0,0 +1,137 @@
+const PersonDetailsContainer = styled.div<{ isHidden?: boolean }>(
+  ({ isHidden }) => {
+    let colors = `
+      background-color: #f0f0f0;
```
minor - could be one line and not a string literal (use `'` not a tick

goralight commented on src/components/Molecules/PersonCard/index.tsx:
```diff
@@ -0,0 +1,137 @@
+    `
+    if (isHidden) {
+      colors = `
+        background-color: #84c489;
```
same here

Noorieee committed:
Adjusted styles to make the PersonCard consistent even when isHidden
Added a render in stories and adjusted the width of the PersonCard
Changed ticks to apostrophes in PersonDetailsContainer
